qiime2/sdk/proxy.py

```diff
@@ -39,21 +44,37 @@ def __init__(self, future, selector, signature=None):
         self._selector_ = selector
         self._signature_ = signature
 
+    def __repr__(self):
+        if self._signature_ is None:
+            return f'<{self.__class__.__name__.__lower__}: Unknown Type ' \
+                   f'{object.__repr__(self)}>'
+        else:
+            return f'<{self.__class__.__name__.__lower__}: {self.type}>'
+
+    def __hash__(self):
+        return hash(self.uuid)
+
     @property
     def _archiver(self):
         return self.result()._archiver
 
     @property
     def type(self):
-        return self._archiver.type
+        if self._signature_ is not None:
+            return self._signature_[self._selector_].qiime_type
```

Comment:
Do we still need to index via the selector here? It seems like we can just have the type as a private attr